pkg/plugins/manager/signature/manifestverifier/verifier.go

```diff
@@ -111,20 +112,68 @@ func (pmv *ManifestVerifier) GetPublicKey(keyID string) (string, error) {
 	}()
 
 	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
-		return "", err
+		return err
 	}
 
 	if len(data.Items) == 0 {
-		return "", errors.New("missing public key")
+		return errors.New("missing public key")
 	}
 
 	for _, key := range data.Items {
-		pmv.publicKeys[key.KeyID] = key
+		err = pmv.kv.Set(ctx, key.KeyID, key.PublicKey)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
we may want to use GetAll to grab all current keys and clean up any keys that exist in the store but aren't in Items, so that any bad entries are removed from the database automatically

Reply:
If we decide to keep track of the last download time, the keys should likely be prefixed (eg with key-) so we can use Keys to get all keys and still be able to store a last updated date.

When we have last updated date we can and should also use that to bootstrap the background update process so that instances that don't run for 10 days continuously will still update their keys after 10 days.  There is an example of how to do that using the kvstore here https://github.com/grafana/grafana/blob/main/pkg/infra/usagestats/service/service.go#L61-L99

Reply:
done both things :+1: